--- a/nuttx/drivers/audio/es8311.c
+++ b/nuttx/drivers/audio/es8311.c
@@ -1048,13 +1048,13 @@
         priv->samprate  = caps->ac_controls.hw[0];
         priv->bpsamp    = caps->ac_controls.b[2];
 
-        ret = es8311_setsamplerate(priv) == -ENOTTY ? OK : ret;
+        ret = es8311_setsamplerate(priv);
         if (ret < 0)
           {
             break;
           }
 
-        ret = es8311_setbitspersample(priv) == -ENOTTY ? OK : ret;
+        ret = es8311_setbitspersample(priv);
       }
       break;
 
@@ -1095,13 +1095,13 @@
         priv->samprate  = caps->ac_controls.hw[0];
         priv->bpsamp    = caps->ac_controls.b[2];
 
-        ret = es8311_setsamplerate(priv) == -ENOTTY ? OK : ret;
+        ret = es8311_setsamplerate(priv);
         if (ret != OK)
           {
             break;
           }
 
-        ret = es8311_setbitspersample(priv) == -ENOTTY ? OK : ret;
+        ret = es8311_setbitspersample(priv);
       }
       break;
 
